project/controller/newRouteController.php

```diff
@@ -1,58 +1,73 @@
 <?php
 
 include_once "../constants/routeConstants.php";
-
 include_once '../utils/sessionUtils.php';
 include_once '../database/routeUtils.php';
 
+loginRequired();
+
+
 $distance= getDataFromPOST('distance');
 $name = getDataFromPOST('routeName');
 $mode = getDataFromPOST('mode');
-$team = getDataFromPOST('teamID');
+$team = getDataFromPOST('team');
 $startLatitude = getDataFromPOST('startLatitude');
 $startLongitude = getDataFromPOST('startLongitude');
 $endLatitude = getDataFromPOST('endLatitude');
 $endLongitude = getDataFromPOST('endLongitude');
 
 $userId = getActiveUserID();
 
-if($userId === null) {
-    //TODO: Session is not initialized, move to login?
-}
-
-if($mode === null) {
-    //TODO: Is this check even needed? Can user somehow hack POST so he does not post one value?
+if (!nullCheck(array($distance, $name, $mode, $team, $startLatitude, $startLongitude, $endLatitude, $endLongitude))) {
+    redirectToHomePageWithMessage(NOT_ENOUGH_DATA);
+    return;
 }
 
+//TODO assign selected Team to Route
 switch ($mode) {
     case PRIVATE_MODE:
-        saveRoute($userId, $name, $distance, $mode,
+        saveRoute_FAKE($userId, $name, $distance, $mode,
             $startLatitude, $startLongitude, $endLatitude, $endLongitude);
         break;
 
     case PUBLIC_MODE:
     case TEAM_MODE:
         if (isUserAdmin()) {
-            saveRoute($userId, $name ,$distance, $mode,
+            saveRoute_FAKE($userId, $name ,$distance, $mode,
                 $startLatitude, $startLongitude, $endLatitude, $endLongitude);
         } else {
-            //TODO: Give error message back?
+            loginRequired(ADMIN_ROLE);
         }
         break;
 
     default:
         break;
 }
 
+redirectToHomePageWithMessage(ROUTE_SUCCESSFULLY_SAVED);
+
+
 function getDataFromPOST($key) {
-    if (isset($_POST[$key])) {
+    if (isset($_POST[$key]) and $_POST[$key] !== "") {
```

Comment:
Time does not come as empty string I think. But as 00:00:00. Can we investigate this more?

Reply:
i will look into this

Reply:
Something like php function empty() maybe?

Reply:
checked it out, both time and date inputs give empty strings in POST when sent empty

Reply:
Any news regarding this? @timbez